ppl/zqd/recruiter/recruiter_test.go

```diff
@@ -0,0 +1,289 @@
+package recruiter
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Register1(t *testing.T, addr string, nodename string, fp int, np int, rp int) *WorkerPool {
+	wp := NewWorkerPool()
+	wp.Register(addr, nodename)
+	AssertPoolLen(t, wp, fp, np, rp)
+	return wp
+}
+
+func AssertPoolLen(t *testing.T, wp *WorkerPool, fp int, np int, rp int) {
+	assert.Equal(t, fp, wp.LenFreePool(), "FreePool len=%d", fp)
+	require.Equal(t, np, wp.LenNodePool(), "NodePool len=%d", np)
+	assert.Equal(t, rp, wp.LenReservedPool(), "ReservedPool len=%d", rp)
+}
+
+func TestBadCalls(t *testing.T) {
+	wp := NewWorkerPool()
+	err := wp.Register("a.b;;5000", "n1")
+	assert.NotNil(t, err)
+	AssertPoolLen(t, wp, 0, 0, 0)
+	err = wp.Register("a.b:5000", "")
+	assert.NotNil(t, err)
+	AssertPoolLen(t, wp, 0, 0, 0)
+}
+
+func TestRegisterTwice(t *testing.T) {
+	wp := Register1(t, "a.b:5000", "n1", 1, 1, 0)
+	wp.Register("a.b:5000", "n1")
+	AssertPoolLen(t, wp, 1, 1, 0)
+}
+
+func TestDeregister1(t *testing.T) {
+	wp := Register1(t, "a.b:5000", "n2", 1, 1, 0)
+	wp.Deregister("a.b:5000")
+	AssertPoolLen(t, wp, 0, 0, 0)
+}
+
+func TestRecruit1(t *testing.T) {
+	addr := "a.b:5000"
+	nodename := "n2"
+	wp := Register1(t, addr, nodename, 1, 1, 0)
+	s, err := wp.Recruit(1)
+	require.Nil(t, err)
```

Comment:
This applies in lots of other locations as well.
```suggestion
	require.NoError(t, err)
```